### PR TITLE
PRELIMINARY REVIEW: 0006-application-management.md

### PR DESCRIPTION
**Pull request by craigmcnally**
Preliminary Review of proposal for Application Management

### CODE REVIEW

**Comment by starsplatter** [comment]
The mention of application stores/marketplaces seems to stand out as non-technical compared to the other scoped items. Is it possible to clarify why about stores/marketplaces will be in the RFC?

**Comment by todolson** [comment]
Following up on @starsplatter's comment about application stores/marketplaces:

Would it be more clear (and still accurate) to say something like "application management is a step towards facilitating deployment of applications from multiple repositories into a single tenant, such as different applications from both project and third-party repositories"? (Regardless, I expect some pushback that the distribution could be handled entirely at the module level.)

It seems like the "Application Manager" function that was mentioned in the original "Application formalization" document was about managing the deployment of a coherent set of modules within a tenant or a running environment, and that the mechanics of managing a marketplace of multiple repositories is a little further down the road. So I wonder how much app store/marketplace info belongs here, or if the focus should be on deploying apps as defined sets of modules.

File: text/0006-application-management.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+## Summary
+The introduction of formalized Applications introduces the need to manage such applications in a reliable fashion.  This would also be the first step in being able to move towards a comprehensive App Store or Marketplace vision. Management starts at the application-level.  These manager components will provide APIs working at the application-level, while consuming the lower level Okapi APIs for module-level, tenant, and tenant entitlement operations.
+
+## Scope
+* Application registration
+* Tenant creation and management
+* Application/Tenant entitlement
+* Application Stores & Marketplaces
+* How movement of modules between Applications will work
```
Comment (marcjohnson-kint) [inline text/0006-application-management.md:22]: Is deployment of modules part of application management?

File: text/0006-application-management.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+# Application Management
+
+## Summary
+The introduction of formalized Applications introduces the need to manage such applications in a reliable fashion.  This would also be the first step in being able to move towards a comprehensive App Store or Marketplace vision. Management starts at the application-level.  These manager components will provide APIs working at the application-level, while consuming the lower level Okapi APIs for module-level, tenant, and tenant entitlement operations.
+
+## Scope
+* Application registration
+* Tenant creation and management
+* Application/Tenant entitlement
```
Comment (marcjohnson-kint) [inline text/0006-application-management.md:20]: What does entitlement mean in this context? Is it similar to module enablement?

Comment (craigmcnally) [reply text/0006-application-management.md:20]: I've added a terminology section and define application entitlement.

File: text/0006-application-management.md (hunk 1 above)
Comment (craigmcnally) [reply text/0006-application-management.md:22]: I've added this to the scope statement.

**Comment by craigmcnally** [comment]
> The mention of application stores/marketplaces seems to stand out as non-technical compared to the other scoped items. Is it possible to clarify what about stores/marketplaces will be in the RFC?

I've added clarification to this item in the scope statement.  In the future, please use inline comments.  These cannot be resolved, nor do they appear in the "files" view.

**Comment by craigmcnally** [comment]
> Following up on @starsplatter's comment about application stores/marketplaces:
> 
> Would it be more clear (and still accurate) to say something like "application management is a step towards facilitating deployment of applications from multiple repositories into a single tenant, such as different applications from both project and third-party repositories"? (Regardless, I expect some pushback that the distribution could be handled entirely at the module level.)
> 
> It seems like the "Application Manager" function that was mentioned in the original "Application formalization" document was about managing the deployment of a coherent set of modules within a tenant or a running environment, and that the mechanics of managing a marketplace of multiple repositories is a little further down the road. So I wonder how much app store/marketplace info belongs here, or if the focus should be on deploying apps as defined sets of modules.

@todolson / @starsplatter I've clarified this item in the scope statement:
`* How this is a step towards Application Stores & Marketplaces`

In the future, please use inline comment threads. Top-level comments cannot be resolved, nor do they appear in the "Files changed" view.

File: text/0006-application-management.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  * PUBLIC REVIEW: TBD
+  * FINAL REVIEW: TBD
+* Outcome: 
+
+# Application Management
+
+## Summary
+The introduction of formalized Applications introduces the need to manage such applications in a reliable fashion.  This would also be the first step in being able to move towards a comprehensive App Store or Marketplace vision. Management starts at the application-level.  These manager components will provide APIs working at the application-level, while consuming the lower level Okapi APIs for module-level, tenant, and tenant entitlement operations.
+
+## Scope
```
Comment (marcjohnson-kint) [inline text/0006-application-management.md:17]: As discussed in today's TC meeting, please include `backward compatibility with module only operation of FOLIO instances` to the scope (only suggested wording)

Comment (craigmcnally) [reply text/0006-application-management.md:17]: I've added a similarly worded bullet to the scope statement

**Pull request closed without merging**